tests/integration/test_micrometer_metrics.py

```diff
@@ -0,0 +1,65 @@
+from tests.integration import basetest
+
+
+class TestMicrometerMetricsFlow(basetest.BaseTestWithPostgreSQL):
+    """Test the metrics flow when metrics via micrometer is enabled.
+
+    Even with micrometer enabled, we still have the following list of metrics
+    pushed to TSS via python metrics emitter:
+    - database
+    - storage
+    - health
+    - smaps
+    - critical log count
+    Here we assume the metrics are written to STDOUT with BYPASS_LOGGREGATOR
+    set to false, which is not the case in production. These tests just ensure
+    the correct metrics are still being emitted.
+
+    Runtime stats like memory, threadpool are pushed to TSS via telegraf,
+    which are not being tested here.
+    """
+
+    def test_free_apps_metrics(self):
+        """Test no metrics for free apps via old stream."""
+        self.stage_container(
+            "BuildpackTestApp-mx9-7.mda",
```

Comment:
can we add a negative test for runtimes < 9.7?

Reply:
For the negative tests (versions <9.7), we already have integration tests [here](https://github.com/mendix/cf-mendix-buildpack/blob/51ec6ebe63680d4e5e58f04301e3b1ab43782fef/tests/integration/test_metrics.py) for freeapps and paid apps. These additional tests are just to ensure the micrometer scenario.